### PR TITLE
[ALP-13] New volunteer form

### PR DESCRIPTION
**Pull request by michaelfromyeg**
Known issues:
- Province field and country field should be select components
- Define constants for the form (everything is currently hardcoded)
- Making the volunteer 'state' fields consistent across codebase (i.e., the fields the form takes do not currently match the constant volunteer fields)
- Form styling
- DOM error creating by textInput component
> Warning: validateDOMNesting(...): <div> cannot appear as a descendant of <p>.
    in div (created by ForwardRef(FormControl))
    in ForwardRef(FormControl) (created by WithStyles(ForwardRef(FormControl)))
    in WithStyles(ForwardRef(FormControl)) (created by ForwardRef(TextField))
    in ForwardRef(TextField) (created by WithStyles(ForwardRef(TextField)))
    in WithStyles(ForwardRef(TextField)) (at textInput.jsx:33)
    in p (created by ForwardRef(Typography))
    in ForwardRef(Typography) (created by WithStyles(ForwardRef(Typography)))
    in WithStyles(ForwardRef(Typography)) (at textInput.jsx:30)
    in div (at textInput.jsx:29)
    in TextInput (at newVolunteer.jsx:97)
    in form (at newVolunteer.jsx:71)
    in div (at newVolunteer.jsx:70)
    in div (at newVolunteer.jsx:67)
    in NewVolunteer (at App.js:12)
    in App (at src/index.js:7)

What works:
- Form is fully built, takes state
- onClick, writes state to console

### CODE REVIEW

**Review by thaingocnguyen — APPROVED** [review]
Aside from the known issues, looks great! I'll look into how we can fix the DOM error.

File: aunt-leahs-app/src/pages/newVolunteer.jsx [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+              <TextInput title="Phone number" size="Short" onChange={(e) => this.handleChange(e, 'contactPhone')} value={this.state.contactPhone} />
+            </div>
+            <TextInput title="Email Address" size="Long" onChange={(e) => this.handleChange(e, 'contactEmail')} value={this.state.contactPhone}/>
+
+            <Tickbox
+              onChange={this.handleTickbox}
+              checked={this.state.mailingList}
+              title="I would like to be added to Aunt Leah’s mailing list." // Should be moved to constants, I'm just not sure how or where
+              color="primary"
+            ></Tickbox>
```
Comment (thaingocnguyen) [inline aunt-leahs-app/src/pages/newVolunteer.jsx]: No children inside so can Tickbox can be self-closing

File: aunt-leahs-app/src/pages/newVolunteer.jsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  handleChange(event, stateToChange) {
+    this.setState({ [stateToChange] : event.target.value });
+  }
+
+  handleTickbox(event) {
+    this.setState({ mailingList : !this.state.mailingList });
+  }
+
+  onClick(event) {
+    console.log(this.state);
```
Comment (codygagnon) [inline aunt-leahs-app/src/pages/newVolunteer.jsx:62]: Should this console log be here?

File: aunt-leahs-app/src/pages/newVolunteer.jsx [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+            <h3 className="new-volunteer-form-subheading">Emergency Contact</h3>
+            <div className="formRow">
+              <TextInput title="First name" size="Short" onChange={(e) => this.handleChange(e, 'contactFName')} value={this.state.contactFName}/>
+              <TextInput title="Last name" size="Short" onChange={(e) => this.handleChange(e, 'contactLName')} value={this.state.contactLName}/>
+            </div>
+            <div className="formRow">
+              <TextInput title="Relationship" size="Short" onChange={(e) => this.handleChange(e, 'contactRelationship')} value={this.state.contactRelationship}/>
+              <TextInput title="Phone number" size="Short" onChange={(e) => this.handleChange(e, 'contactPhone')} value={this.state.contactPhone} />
+            </div>
+            <TextInput title="Email Address" size="Long" onChange={(e) => this.handleChange(e, 'contactEmail')} value={this.state.contactPhone}/>
```
Comment (codygagnon) [inline aunt-leahs-app/src/pages/newVolunteer.jsx:97]: Michael, this is fine for now, but you could write code that is far less repetitive if you used Material UI's Grid component and iterated over a form array that contains an object for each field. This form array could be imported from the constants file. Have a look at the docs for the Grid component here: https://material-ui.com/components/grid/ and here: https://material-ui.com/api/grid/. Consider making a ticket to refactor this code.

File: aunt-leahs-app/src/App.js [hunk 4]
```diff
@@ -1,13 +1,16 @@
...
 
 
 function App() {
   return (
-    <div className='testtest'>
-        <AdminLogin />
-    </div>
+    <>
+      <HomePage />
+      <NewVolunteer></NewVolunteer>
```
Comment (codygagnon) [inline aunt-leahs-app/src/App.js]: This should be a self-closing tag.

File: aunt-leahs-app/src/pages/newVolunteer.jsx [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+      contactEmail: "",
+      mailingList: false,
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleTickbox = this.handleTickbox.bind(this);
+    this.onClick = this.onClick.bind(this);
+  }
+
+  handleChange(event, stateToChange) {
+    this.setState({ [stateToChange] : event.target.value });
```
Comment (codygagnon) [inline aunt-leahs-app/src/pages/newVolunteer.jsx]: Rather than stateToChange, you might name the variable field, fieldName, or key to improve readability.

**Review by codygagnon — CHANGES_REQUESTED** [review]
Michael, please take a look at my comments. I've suggested just a few small changes.

**Comment by codygagnon** [comment]
Conflicts need to be resolved before merging.

**Comment by thaingocnguyen** [comment]
> Conflicts need to be resolved before merging.

The conflicts are resolved now, Cody just need to approve the changes or Michael needs to make the changes and resolve them.

**Comment by codygagnon** [comment]
The last PR that was merged has caused a couple more merge conflicts. Can someone resolve these?

**Comment by thaingocnguyen** [comment]
> The last PR that was merged has caused a couple more merge conflicts. Can someone resolve these?
Okay let me resolve these. I will add the temp. routing for this page for demo purposes as well.

File: aunt-leahs-app/src/pages/newVolunteer.jsx (hunk 2 above)
Comment (thaingocnguyen) [reply aunt-leahs-app/src/pages/newVolunteer.jsx:62]: I think this is just temporary until we implementing the saving logic

**Review by codygagnon — APPROVED** [review]

**Pull request merged**